pkg/flink/internal/controller/statement_controller_test.go

```diff
@@ -312,6 +312,27 @@ func (s *StatementControllerTestSuite) TestExecuteStatementReturnsWhenUserDetach
 	cupaloy.SnapshotT(s.T(), stdout)
 }
 
+func (s *StatementControllerTestSuite) TestExecuteStatementWithWarning() {
+	statementToExecute := "insert into users values ('test');"
+	warning := "[Warning] The primary key does not match the upsert key derived from the query. If the primary key and upsert key don't match, the system needs to add a state-intensive operation for correction. Please revisit the query (upsert key: [customer_name]) or the table declaration for `default`.`dmvk`.`o` (primary key: [total_orders])"
```

Comment:
To match the style of the other warnings in the CLI:
```suggestion
	warning := "[WARN] The primary key does not match the upsert key derived from the query. If the primary key and upsert key don't match, the system needs to add a state-intensive operation for correction. Please revisit the query (upsert key: [customer_name]) or the table declaration for `default`.`dmvk`.`o` (primary key: [total_orders])"
```

Reply:
This is just a mock text for the unit test, not what we'll actually show.

The actual warning comes out of Flink and is used across all clients, and we can't change the text for it.